cmd/rookctl/pool/create.go

```diff
@@ -82,31 +110,36 @@ func createPoolsEntry(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func createPool(poolName, poolType string, replicaCount, dataChunks, codingChunks uint, c client.RookRestClient) (string, error) {
-	if poolType != PoolTypeReplicated && poolType != PoolTypeErasureCoded {
-		return "", fmt.Errorf("invalid pool type '%s', allowed pool types are '%s' and '%s'",
-			poolType, PoolTypeReplicated, PoolTypeErasureCoded)
+func ConfigToModel(config Config) (*model.Pool, error) {
+
+	if config.PoolType != PoolTypeReplicated && config.PoolType != PoolTypeErasureCoded {
+		return nil, fmt.Errorf("invalid pool type '%s', allowed pool types are '%s' and '%s'",
+			config.PoolType, PoolTypeReplicated, PoolTypeErasureCoded)
 	}
 
-	newPool := model.Pool{Name: poolName}
+	newPool := &model.Pool{}
 
-	if poolType == PoolTypeReplicated {
-		if dataChunks > 0 || codingChunks > 0 {
-			return "", fmt.Errorf("both data chunks and coding chunks must be zero for pool type '%s'", PoolTypeReplicated)
+	if config.PoolType == PoolTypeReplicated {
+		if config.DataChunks > 0 || config.CodingChunks > 0 {
```

Comment:
nit: Use != 0 otherwise negative numbers will make this condition to work

Reply:
negative numbers could be ignored just like 0